### PR TITLE
add HA sg repair

### PR DESCRIPTION
**Pull request by bzhaoopenstack**
This PR add a ha check cli for checking existing node SG configuration.

summary:
openlab ha cluster repair --security-group
This cmd will try to check and repair the SG if find errors.

openlab ha cluster repair --security-group --dry-run
This cmd will only check the SG of existing nodes.

If we launch a new HA deployment, such as new-slave. We can just run this cmd(if the deployment cloud already contains SG 'openlab-ha-ports'), it will fix the all ha related SG rules automatically.

Related-Bug: theopenlab/openlab#218

### CODE REVIEW

File: openlabcmd/openlabcmd/cli.py [hunk 1]
```diff
@@ -187,6 +187,19 @@ def _add_ha_cluster_cmd(self, parser):
             'switch', help='Switch Master and Slave role.')
         cmd_ha_service_get.set_defaults(func=self.ha_cluster_switch)
 
+        # openlab ha cluster check
+        cmd_ha_cluster_check = cmd_ha_cluster_subparsers.add_parser(
+            'check', help='HA deployment check.')
```
Comment (wangxiyuan) [inline openlabcmd/openlabcmd/cli.py]: So actually this command does `check` and `fix`, maybe the word `check` here is a little misleading. How about `openlab ha cluster repair`?

AND the  help message should contain more detail. at least it should contain what will be done?

File: openlabcmd/openlabcmd/cli.py [hunk 2]
```diff
@@ -391,6 +404,16 @@ def ha_cluster_switch(self):
             print("Switch success")
         except exceptions.OpenLabCmdError:
             print("Switch failed")
+    
+    @_zk_wrapper
+    def ha_cluster_check(self):
+        # TODO(bz) This check may support other function
```
Comment (wangxiyuan) [inline openlabcmd/openlabcmd/cli.py]: ++

File: openlabcmd/openlabcmd/zk.py [hunk 3]
```diff
@@ -30,6 +32,17 @@
...
+RSYNCD_HA_PORT = 873
+ZOOKEEPER_HA_PORTS = [2181, 2888, 3888]
+MYSQL_HA_PORT = 3306
+
+HA_PORTS = []
+for ports in [RSYNCD_HA_PORT, ZOOKEEPER_HA_PORTS, MYSQL_HA_PORT]:
+    if isinstance(ports, list):
+        HA_PORTS.extend(ports)
+    else:
+        HA_PORTS.append(ports)
```
Comment (wangxiyuan) [inline openlabcmd/openlabcmd/zk.py]: Please move L35-L44 to a new place, you can even create a new file call `repair.py`. Don't use global parameters by default.

File: openlabcmd/openlabcmd/cli.py (hunk 1 above)
Comment (bzhaoopenstack) [reply openlabcmd/openlabcmd/cli.py]: Yeah, I agree with u. This cmd can check and fix. Will add in the next commit

File: openlabcmd/openlabcmd/zk.py (hunk 3 above)
Comment (bzhaoopenstack) [reply openlabcmd/openlabcmd/zk.py]: Right, lets add a new file, even named a 'common.py' to store these values.

File: openlabcmd/openlabcmd/cli.py [hunk 4]
```diff
@@ -187,6 +187,19 @@ def _add_ha_cluster_cmd(self, parser):
             'switch', help='Switch Master and Slave role.')
         cmd_ha_service_get.set_defaults(func=self.ha_cluster_switch)
 
+        # openlab ha cluster repair
+        cmd_ha_cluster_check = cmd_ha_cluster_subparsers.add_parser(
```
Comment (wangxiyuan) [inline openlabcmd/openlabcmd/cli.py]: `cmd_ha_cluster_check` should be changed as well.

**Review by wangxiyuan — COMMENTED** [review]
a nit

**Review by wangxiyuan — APPROVED** [review]

**Pull request merged**